spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/DeletedColumnVector.java

```diff
@@ -28,11 +27,14 @@
 import org.apache.spark.unsafe.types.UTF8String;
 
 public class DeletedColumnVector extends ColumnVector {
-  private final boolean[] isDeleted;
```

Comment:
I am not super excited to make it mutable. We would not have to do this if we loaded metadata vectors after data vectors and computing `isDeleted`. That said, let me think a bit more tomorrow.

Reply:
will keep this for now and reconsider this later